src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js

```diff
@@ -55,33 +61,49 @@ class ActivityFeed extends React.Component<Props, State> {
 
     componentDidUpdate(prevProps: Props, prevState: State) {
         const { feedItems: prevFeedItems } = prevProps;
-        const { feedItems: currFeedItems } = this.props;
+        const { feedItems: currFeedItems, activeFeedEntryId } = this.props;
         const { isInputOpen: prevIsInputOpen } = prevState;
         const { isInputOpen: currIsInputOpen } = this.state;
 
-        const isEmpty = this.isEmpty(this.props);
-        const wasEmpty = this.isEmpty(prevProps);
-        const hasLoaded = isEmpty !== wasEmpty && !isEmpty;
-
+        const hasLoaded = this.hasLoaded();
         const hasMoreItems = prevFeedItems && currFeedItems && prevFeedItems.length < currFeedItems.length;
-        const hasNewItems = !prevFeedItems && currFeedItems;
+        const didLoadFeedItems = prevFeedItems === undefined && currFeedItems !== undefined;
         const hasInputOpened = currIsInputOpen !== prevIsInputOpen;
 
-        if (hasLoaded || hasMoreItems || hasNewItems || hasInputOpened) {
+        if ((hasLoaded || hasMoreItems || didLoadFeedItems || hasInputOpened) && activeFeedEntryId === undefined) {
             this.resetFeedScroll();
         }
+
+        // do the scroll only once after first fetch of feed items
+        if (didLoadFeedItems) {
+            this.scrollToActiveFeedItem();
+        }
+    }
+
+    scrollToActiveFeedItem() {
+        const { current: activeFeedItemRef } = this.activeFeedItemRef;
+        const { activeFeedEntryId } = this.props;
+        if (!activeFeedEntryId) {
+            return;
+        }
+
+        scrollIntoView(activeFeedItemRef);
     }
 
     /**
      * Detects whether or not the empty state should be shown.
      * @param {object} currentUser - The user that is logged into the account
      * @param {object} feedItems - Items in the activity feed
      */
-    isEmpty = ({ currentUser, feedItems }: Props = this.props): boolean =>
-        !currentUser ||
-        !feedItems ||
-        feedItems.length === 0 ||
-        (feedItems.length === 1 && feedItems[0].type === ItemTypes.fileVersion);
+    isEmpty = ({ feedItems }: Props = this.props): boolean => {
+        if (feedItems === undefined) {
+            return false;
+        }
+        return feedItems.length === 0 || (feedItems.length === 1 && feedItems[0].type === ItemTypes.fileVersion);
+    };
+
+    hasLoaded = ({ currentUser, feedItems }: Props = this.props): boolean =>
```

Comment:
Minor: Should we make this `isLoading`, since that's how it seems to be used most often (in render)?

Reply:
@tjgupta  and I discussed it and we think sticking to having both hasLoaded and isLoading is ideal since we are using both within the file